util/src/main/java/org/folio/reservoir/util/oai/OaiHttpRequest.java

```diff
@@ -0,0 +1,127 @@
+package org.folio.reservoir.util.oai;
+
+import io.netty.handler.codec.http.QueryStringEncoder;
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.RequestOptions;
+import io.vertx.core.streams.ReadStream;
+import java.util.HashMap;
+import java.util.Map;
+import org.folio.reservoir.util.XmlMetadataStreamParser;
+import org.folio.reservoir.util.readstream.XmlFixer;
+import org.folio.reservoir.util.readstream.XmlParser;
+
+public class OaiHttpRequest<T> implements OaiRequest<T> {
```

Comment:
Rename to "ListRecordsRequestImpl"

Reply:
No. There is going to be one for Storage and Postgresql later

Reply:
Makes sense, then rename this to `ListRecordsRequestOai` and the interface to `ListRecordsRequest`, same for 
`*Response` ones. The Postgres implementation can be called `ListRecordsRequestPg` or similar.

I now get what you're after so I propose that we use the term `ListRecords` to refer to the abstract interface for which OAI-PMH is a concrete wire-level implementation. 

We could even get crazy and roll out our own wire-level implementation which is JSON-based or content-type agnostic and uses HTTP headers for protocol-level metadata like resumption tokens.